lib/scenes/level_scene.cpp

```diff
@@ -117,6 +119,36 @@ void LevelScene::performPrepare() {
         entity_components->push_back(factory.createImage("advertisement/billboard.png", x, y, billboard_x_scale, billboard_y_scale, getRelativeModifier(), Layers::Lowground, alpha));
         entity_components->push_back(factory.createImage(content_path, x, y, content_x_scale, content_y_scale, getRelativeModifier(), Layers::Lowground, alpha));
     }
+
+    // Create HUD Components
+    auto player_entities = factory.getEntityManager().getEntitiesByComponent<PlayerComponent>();
+
+    int spacing = screen_width / (player_entities.size() + 1);
+    
+    std::vector<int> player_ids;
+    for (auto& [entity_id, player] : player_entities) {
+        player_ids.push_back(player->player_id);
+    }
+    sort(player_ids.begin(), player_ids.end());
+
+    for (auto& [entity_id, player] : player_entities) {
+        auto hud_component = factory.getEntityManager().getComponent<HUDComponent>(entity_id);
+
+        std::vector<int>::iterator it = std::find(player_ids.begin(), player_ids.end(), player->player_id);
+        int index = std::distance(player_ids.begin(), it);
```

Comment:
This construction has quite a high chance of crashing, because you don't check whether there is a nullptr.
Also put auto instead of that ugly iterator.